### DISCUSSION TITLE
Sioyek not rendering pdf contents

### DISCUSSION THREAD

**Discussion by javierpelayo** [ahrm/sioyek#475] [Q&A] [answered]
I've tried all versions of sioyek and building from source as well as installing through the AUR. I'm using Arch linux. I've also downgraded from libmupdf 1.20 to 1.19 but the issue persisted, I'm thinking it has to do with either sioyek itself or one of its dependencies such as qt5-3d. Everytime I open a pdf file the pages themselves are blank i.e. just a white background. Table of contents works and the page numbers are correct/everything else works except for the fact that the pdf pages are blank. I have tried testing out searching for a string of characters as well and I've noticed that the page quickly flashes with highlighted characters but then they disappear into the white background.

**Comment by ahrm** [comment, marked as answer]
Perhaps that's an OpenGL issue? Make sure your graphics driver supports at least OpenGL 3.3.

**Reply by ntevenhere** [reply]
I have this issue. What if my graphics driver doesn't support OpenGL 3? Decade-old computers like the thinkpad x200 don't support newer OpenGL versions. Is Sioyek simply incompatible?

**Reply by ahrm** [reply]
I think there are mesa software drivers available for opengl 3.3 .

**Reply by ntevenhere** [reply]
> I think there are mesa software drivers available for opengl 3.3 .

I have mesa installed. `glxinfo | grep -i opengl` is still `OpenGL version string: 2.1`. This was the roadblock I came to when I tried to try sioyek months ago. 

I did some research now and actually found `LIBGL_ALWAYS_SOFTWARE=1` It works!! I can finally see text in Sioyek! On the other hand this a variable you usually don't want to enable since it degrades performance for the type of performance critical software that asks for OpenGL, games, modeling software etc. Software rendering on Sioyek seems to not be that slow but I have to use it more to say for sure.
